Bottom: 9fddb69173a0a10d4cd7464edc4d87623f9c2d01
Top:    26d09955dd68a4054cfb99a15ca38652f6369464
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 13:19:56 +0000

Try not using no-holes dicer for drawing polygons









---

diff --git a/src/draw.c b/src/draw.c
index 70c6223..60e1175 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -2254,7 +2254,7 @@ DrawPour (LayerTypePtr Layer, PourTypePtr Pour, int unused)
   layernum = GetLayerNumber (PCB->Data, Layer);
   DrawPourLowLevel (Pour, NULL);
 #warning FIXME Later
-#if 0
+#if 1
   if (TEST_FLAG (CLEARPOLYFLAG, Pour))
     {
       r_search (PCB->Data->pin_tree, &Pour->BoundingBox, NULL,
@@ -2322,6 +2322,7 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
     }
   else if (Polygon->Clipped)
     {
+#if 0
       if (!Polygon->NoHolesValid)
         {
           ComputeNoHoles (Polygon);
@@ -2335,6 +2336,15 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
             poly.Clipped = poly.Clipped->f;
           } while (poly.Clipped != Polygon->NoHoles);
         }
+#endif
+        {
+          PolygonType poly = *Polygon;
+          poly.Clipped = Polygon->Clipped;
+          do {
+            DrawPolygonLowLevel (&poly, NULL);
+            poly.Clipped = poly.Clipped->f;
+          } while (poly.Clipped != Polygon->Clipped);
+        }
       /* draw other parts of the polygon if fullpoly flag is set */
       /* NB: No "NoHoles" cache for these */
       if (TEST_FLAG (FULLPOLYFLAG, Polygon))
